lib/vfscore/include/vfscore/vnode.h

```diff
@@ -43,6 +43,8 @@
 #include <vfscore/uio.h>
 #include <vfscore/dentry.h>
 
+#define IOCTL_CMD_ISTTY(cmd)		((cmd & ('T' << 8)) == ('T' << 8))
```

Comment:
Maybe let's make it even more generic? How about this
```C
#define IOCTL_CMD_TYPE_SHIFT		    8
#define IOCTL_CMD_TYPE_MASK		    (0xFF << IOCTL_CMD_TYPE_SHIFT)
#define IOCTL_CMD_TYPE_TTY                     ('T')
#define IOCTL_CMD_ISTYPE(cmd, type)     (((cmd) & IOCTL_CMD_TYPE_MASK) == (type << IOCTL_CMD_TYPE_SHIFT))
```
and then
```C
	if (IOCTL_CMD_ISTYPE(com, IOCTL_CMD_TYPE_TTY))
		return ENOTTY;
```
Let me know your thoughts

Reply:
Nice, done.